tests/test_project_busy.py

```diff
@@ -0,0 +1,123 @@
+    # Unit Personnel
+    token = tests.UserAuth(tests.USER_CREDENTIALS["unituser"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+    )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+    assert "Required data missing" in response.json.get("message")
+
+    # Unit Personnel
+    token = tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+    )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+    assert "Required data missing" in response.json.get("message")
+
+    # Researcher
+    token = tests.UserAuth(tests.USER_CREDENTIALS["researchuser"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+    )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+    assert "Required data missing" in response.json.get("message")
+
+
+def test_set_busy_no_busy(client):
+    """busy bool required."""
+    token = tests.UserAuth(tests.USER_CREDENTIALS["projectowner"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+        query_string=proj_query,
+        json={"something": "notabool"},
+    )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+    assert "Are you trying to set the project as busy or not busy?" in response.json.get("message")
+
+
+def test_set_busy_true(client):
+    """Set project as busy."""
+    token = tests.UserAuth(tests.USER_CREDENTIALS["projectowner"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+        query_string=proj_query,
+        json={"busy": True},
+    )
+    assert response.status_code == http.HTTPStatus.OK
+    assert f"Project {proj_query.get('project')} was set to busy." in response.json.get("message")
+
+
+def test_set_busy_false(client):
+    """Set project as not busy."""
+
+    token = tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+        query_string=busy_proj_query,
+        json={"busy": False},
+    )
+    assert response.status_code == http.HTTPStatus.OK
+    assert f"Project {busy_proj_query.get('project')} was set to not busy." in response.json.get(
+        "message"
+    )
+
+
+def test_set_busy_project_already_busy(client):
+    """Set a busy project as busy."""
+
+    token = tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client)
+    response = client.put(
+        tests.DDSEndpoint.PROJECT_BUSY,
+        headers=token,
+        query_string=busy_proj_query,
+        json={"busy": True},
+    )
+    # assert response.status_code == http.HTTPStatus.OK
+    assert "The project is already busy, cannot proceed." in response.json.get("message")
```

Comment:
After this we can also add a new test with the same test except regarding "not busy". So basically just copy the test above and change to not busy.